File: java/IEC61970/Base/ControlArea/AltTieMeas.java
package IEC61970.Base.ControlArea;

import IEC61970.Base.Domain.Integer;
import IEC61970.Base.Meas.AnalogValue;
import IEC61970.Base.Core.IdentifiedObject;

/**
 * A prioritized measurement to be used for the tie flow as part of the control
 * area specification.
 * @author kdd
 * @version 1.0
 * @created 25-Dec-2023 8:31:54 PM
 */
public class AltTieMeas extends IdentifiedObject {

	/**
	 * Priority of a measurement usage.   Lower numbers have first priority.
	 */
	public Integer priority;
	/**
	 * The specific analog value used as a source.
	 */
	public AnalogValue AnalogValue;

	public AltTieMeas(){

	}
}//end AltTieMeas

File: py/IEC61970/Base/ControlArea/AltTieMeas.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Fri Dec 15 17:02:59 2023
from typing import Optional

from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC61970.Base.Meas.AnalogValue import AnalogValue


class AltTieMeas(IdentifiedObject):
    """
    A prioritized measurement to be used for the tie flow as part of the control
    area specification.
    """

    def __init__(self) -> None:
        super().__init__()
        self.priorityOptional[int] = 0  # Priority of a measurement usage. Lower numbers have first priority.
        self.analog_valueOptional[AnalogValue] = AnalogValue()  # The specific analog value used as a source.

